src/evaluator/getters.rs

```diff
@@ -0,0 +1,68 @@
+use sha2::{Digest, Sha256};
+
+use super::models::ConfigSpecType;
+
+pub fn get_hash(s: String) -> u64 {
+    let mut hasher = Sha256::new();
+    hasher.update(s);
+    let res = hasher.finalize();
+    // TODO: Use from_le_bytes
+    (*res.get(7).unwrap_or(&0) as u64)
+        | (*res.get(6).unwrap_or(&0) as u64) << 8
+        | (*res.get(5).unwrap_or(&0) as u64) << 16
+        | (*res.get(4).unwrap_or(&0) as u64) << 24
+        | (*res.get(3).unwrap_or(&0) as u64) << 32
+        | (*res.get(2).unwrap_or(&0) as u64) << 40
+        | (*res.get(1).unwrap_or(&0) as u64) << 48
+        | (*res.first().unwrap_or(&0) as u64) << 56
+}
+
+pub fn get_numeric_value(v: &serde_json::Value) -> Option<f64> {
+    match v {
+        serde_json::Value::Null => None,
+        serde_json::Value::Bool(_) => None,
+        serde_json::Value::Number(n) => n.as_f64(),
+        serde_json::Value::String(s) => s.parse().ok(),
+        serde_json::Value::Array(_) => None,
+        serde_json::Value::Object(_) => None,
+    }
+}
+
+pub fn get_string(v: &serde_json::Value) -> String {
+    match v {
+        serde_json::Value::Null => "".to_string(),
+        serde_json::Value::Bool(b) => b.to_string(),
+        serde_json::Value::Number(n) => n.to_string(),
+        serde_json::Value::String(s) => s.clone(),
+        serde_json::Value::Array(_) => "".to_string(),
+        serde_json::Value::Object(_) => "".to_string(),
+    }
+}
+
+pub fn get_unix_epoch(v: &serde_json::Value) -> i64 {
```

Comment:
Não faz sentido usar unsigned's aqui? epoch é sempre > 0 né?

Reply:
As far as I understand, the epoch can be negative to represent times before 1970. So it should be a signed integer. And that's why it's going to overflow in 2038, because it's a signed 32-bit representation.